client/webserver/site/src/js/forms.ts

```diff
@@ -1175,20 +1193,24 @@ export class DEXAddressForm {
         page.err.textContent = res.msg
         Doc.show(page.err)
       }
-
       return
     }
-
-    if (res.paid) {
+    if (!this.dexToUpdate && res.paid) {
       await app().fetchUser()
       app().loadPage('markets')
       return
     }
-
     if (this.pwCache) this.pwCache.pw = pw
     this.success(res.xc, cert)
   }
 
+  makeDexUpdater (dexToUpdate: string) {
```

Comment:
Please comment here that this is a one way function. The changes made during `makeDexUpdater` are not reversed upon `refresh`. So the same `DEXAddressForm` can't do double duty for new registration and host name updates.


I would actually prefer that `makeDexUpdater` was used internally based on an argument to the constructor.

Reply:
Remove `makeDexUpdater` now?  I believe you have it done in `if (dexToUpdate) {` now.